ext/iperf3/CMakeLists.txt

```
@@ -0,0 +1,32 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# CMakeLists.txt - iperf3 NRF integration
+
+zephyr_library()
+zephyr_include_directories(.) 
```

```suggestion
zephyr_include_directories(.)
```

also, should this be `zephyr_library_include_directories()` ?

---

If I do that change, modem shell won't see `iperf3_api.h`. What would be correct way to make that header available for the clients? `ext/cjson`, which is the only ext library before our PR, has this same style to publish the header to users. If there's a better one, just let us know.

---

> If I do that change, modem shell won't see iperf3_api.h.

Yes, but only because you place `<somefile>.c` that needs `iperf3_api.h` i the default `app` lib.
Let's say the user decides to do:
```
add_library(network STATIC <somefile>.c)
target_link_libraries(app PRIVATE network)
```

then `network` will not be able to obtain the header, and in this case `app` doesn't need the header.

In this case you should create a `iperf3` interface library, like:
```
zephyr_interface_library_named(iperf3)
target_include_directories(iperf3 INTERFACE .....)
```
and in Kconfig:
```
config APP_LINK_WITH_IPERF3
	bool "Link 'app' with iperf3"
	default y
	help
	  .....
```
that will default link the `app` library with `iperf3`, but a user who might want to organize their application differently, as I showed above, can then do:
```
add_library(network STATIC <somefile>.c)
target_link_libraries(network PRIVATE iperf3)
target_link_libraries(app PRIVATE network)
```
and things will keep working, and the scoping of iperf3 includes is limited to where they are needed.

Some background info:
https://github.com/zephyrproject-rtos/zephyr/blob/4dd49ac746dca7762707fe2ae21e4843f09f409d/cmake/extensions.cmake#L544-L568

And an example from mbedTLS:
https://github.com/zephyrproject-rtos/zephyr/blob/4dd49ac746dca7762707fe2ae21e4843f09f409d/modules/mbedtls/CMakeLists.txt#L2-L12
https://github.com/zephyrproject-rtos/zephyr/blob/4dd49ac746dca7762707fe2ae21e4843f09f409d/modules/mbedtls/Kconfig#L154-L160

---

I've changed IPERF3 also to use same style as curl, i.e. `APP_LINK_WITH_IPERF3`